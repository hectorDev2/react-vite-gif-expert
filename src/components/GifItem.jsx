import PropTypes from 'prop-types' // ES6

export const GifItem = ({ title, url, id }) => {
  return (
    <div className='card' aria-label='card'>
      <img src={url} alt={title} />
      <h3 data-testid='title'>{title ? title : ''}</h3>
    </div>
  )
}
GifItem.propTypes = {
  title: PropTypes.string,
  url: PropTypes.string.isRequired,
  id: PropTypes.string.isRequired
}
