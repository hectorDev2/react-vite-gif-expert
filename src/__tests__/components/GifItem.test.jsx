import { GifItem } from '../../components/GifItem'
const { render, screen } = require('@testing-library/react')
describe('test GifItem', () => {
  test('get a snapshots', () => {
    const { container } = render(
      <GifItem
        title='weed is love'
        url='XXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXX'
        id='123'
      />
    )

    expect(container).toMatchSnapshot()
  })

  test('check title and image render', () => {
    // Arrange
    const props = {
      title: 'Test Title',
      url: 'https://example.com/test.gif',
      id: '123'
    }

    // Act
    render(<GifItem {...props} />)

    // Assert
    // expect(screen.getByAltText('Test Title')).()
    expect(screen.getByAltText('Test Title')).toBeTruthy()
    expect(screen.getByRole('img').getAttribute('alt')).toBe(props.title)
  })

  // test('should not throw an error when id is null or undefined', () => {
  //   const props = {
  //     title: 'Test Title',
  //     url: 'https://example.com/test.gif',
  //     id: null
  //   }

  //   render(<GifItem {...props} />)
  // })

  test('should render a p tag with empty text', () => {
    const props = {
      title: null,
      url: 'https://example.com/test.gif',
      id: '1234 '
    }

    render(<GifItem {...props} />)
    expect(screen.getByTestId('title').innerHTML).toBe('')
  })
})
