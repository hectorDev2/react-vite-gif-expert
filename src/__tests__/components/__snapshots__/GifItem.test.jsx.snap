// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`test GifItem get a snapshots 1`] = `
<div>
  <div
    aria-label="card"
    class="card"
  >
    <img
      alt="weed is love"
      src="XXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXX"
    />
    <h3
      data-testid="title"
    >
      weed is love
    </h3>
  </div>
</div>
`;
